sstables/generation_type.hh

```diff
@@ -35,7 +37,14 @@ public:
     constexpr int_t value() const noexcept { return _value; }
 
     explicit operator data_value() const noexcept {
-        return data_value(_value);
+        return data_value(
+            utils::UUID(
+                utils::UUID_gen::create_time(std::chrono::milliseconds::zero()),
+                _value));
+    }
+    static generation_type from_uuid(utils::UUID value) {
+        assert(value.timestamp() == 0);
+        return generation_type(value.get_least_significant_bits());
```

Comment:
This needs an inline comment explaining what you're doing.
Also, why not implement the uuid-generation case upfront?

Reply:
> This needs an inline comment explaining what you're doing. 

@bhalevy added inline comment to explain the rationales.

> Also, why not implement the uuid-generation case upfront?

i want to reduce the mental overhead of reviewers as much as possible. so i can get fast feed back in a timely manner. with these preparing changes landed,  i will have only 4 commits left in my local branch

1. add the feature bits, 
2. add config option
3. use `variant<UUID, int64_t>` for the generation identifier
4. pass `uuid_sstable_identifiers` to generation generator

if i implement the uuid-generation case (the 3rd commit) first, i would have to include all them in one go, as, otherwise the 3rd commit itself does not make sense at all. i am afraid that would be slower to review them in a single PR.

Reply:
I agree, it is better to split work into smaller PRs. A detail like this may have been lost in a larger PR doing a lot of other stuff.